chain-vote/src/lib.rs

```diff
@@ -121,15 +128,25 @@ impl EncryptedTally {
         }
     }
 
-    pub fn finish(&self, secret_key: &OpeningVoteKey) -> (TallyState, TallyDecryptShare) {
+    pub fn finish<R: RngCore + CryptoRng>(
+        &self,
+        rng: &mut R,
+        secret_key: &OpeningVoteKey,
+    ) -> (TallyState, TallyDecryptShare) {
         let mut dshares = Vec::with_capacity(self.r.len());
         let mut r2s = Vec::with_capacity(self.r.len());
         for r in &self.r {
             let (r1, r2) = r.elements();
-            dshares.push(r1 * &secret_key.0.sk);
+            let decrypted_share = r1 * &secret_key.0.sk;
```

Comment:
This is also computed within the `generate()` function. We could pass it as input to avoid computing it twice.